### PR TITLE
Minor stepper refactors and norm metrics fix 

### PR DESCRIPTION
**Pull request by suryadheeshjith**
- Fix the calculation of metrics for normalized data. We were still considering 0s at land.
- Renamed output types and moved them to output.py

### CODE REVIEW

File: src/ocean_emulators/aggregator/inference/main.py [hunk 1]
```diff
@@ -217,147 +223,6 @@ def _get_inference_logs_slice(self, step_slice: slice):
         return to_inference_logs(logs)
 
 
-class InferenceAggregator:
```
Comment (suryadheeshjith) [inline src/ocean_emulators/aggregator/inference/main.py:220]: Removed this class since we are not using it anywhere. Might add it in the future if we do need it!

File: src/ocean_emulators/aggregator/train.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def __init__(self):
         self._loss_per_channel = torch.tensor(torch.nan)
 
     @torch.no_grad()
-    def record_batch(self, batch: TrainOutput):
+    def record_batch(self, batch: TrainStepOutput):
```
Comment (jder) [inline src/ocean_emulators/aggregator/train.py]: :sheep: these names make me think that this should be record_step taking `step: TrainStepOutput`? Or the type should be `TrainBatchOutput`?

File: src/ocean_emulators/utils/data.py [hunk 3]
```diff
@@ -198,7 +199,9 @@ def get_norm_unnorm_dicts(
         data_reshaped = rearrange(data, "n (hi c) h w -> n hi c h w", hi=hist + 1)
 
     # Get normalized dict
-    data_dict = convert_tensor_out_to_dict(data_reshaped)
+    data_normalized = data_reshaped.clone()
```
Comment (jder) [inline src/ocean_emulators/utils/data.py:211]: Why do we need the clone? I don't see any mutations to `data_normalized` or `data_reshaped`.

File: src/ocean_emulators/utils/data.py [hunk 4]
```diff
@@ -198,7 +199,9 @@ def get_norm_unnorm_dicts(
...
+    data_normalized = torch.where(wet == 0, float("nan"), data_normalized)
```
Comment (jder) [inline src/ocean_emulators/utils/data.py:212]: I'm a bit confused why this method needs to do this. It's converting from normalized-data to normalized-data-with-nans-for-land? Is this because this method is only intended to be used by aggregators that want to compute stats across wet cells? Would maybe be nice to try and distinguish that at a type level someday but maybe for now this method could be `get_norm_unnorm_analysis_dicts` or something? Not sure "analysis" is the right name.

File: src/ocean_emulators/eval.py [hunk 5]
```diff
@@ -122,7 +125,9 @@ def __init__(self, cfg: EvalConfig) -> None:
         self.wet, self.wet_surface = extract_wet_mask(
             self.data, self.prognostic_var_names, cfg.data.hist
         )
-        wet_without_hist, _ = extract_wet_mask(self.data, self.prognostic_var_names, 0)
+        self.wet_without_hist, _ = extract_wet_mask(
+            self.data, self.prognostic_var_names, 0
+        )
```
Comment (jder) [inline src/ocean_emulators/eval.py:131]: Why did you want to assign this to self here? It feels worse to me because `self.wet_without_history` is sometimes on self.device and sometimes not. If it's important that Normalize is handed the non-device wet mask then I think we should have this be a local `wet_without_history_cpu` or something. If it's not important, maybe do the `to(self.device)` here?

File: src/ocean_emulators/train.py [hunk 6]
```diff
@@ -200,7 +200,9 @@ def __init__(self, cfg: TrainConfig) -> None:
         self.wet, self.wet_surface = extract_wet_mask(
             self.data, self.prognostic_var_names, cfg.data.hist
         )
-        wet_without_hist, _ = extract_wet_mask(self.data, self.prognostic_var_names, 0)
+        self.wet_without_hist, _ = extract_wet_mask(
+            self.data, self.prognostic_var_names, 0
+        )
```
Comment (jder) [inline src/ocean_emulators/train.py:205]: Same comment/question here about overwriting self.wet_without_hist

File: tests/test_utils_data.py [hunk 7]
```diff
@@ -166,7 +170,7 @@ def normalize_input():
             boundary_var_names=["var_2"],
             wet_mask=wet_mask,
         )
-    return normalize, wet_mask
+        yield normalize, wet_mask
```
Comment (jder) [inline tests/test_utils_data.py:173]: Nice catch

File: tests/test_utils_data.py [hunk 8]
```diff
@@ -186,3 +190,104 @@ def test_unnormalize_prognostic_tensor(normalize_input, fill_value):
...
+        tensor_map = TensorMap.init_instance("thetao_surface", "hfds")
+
+        wet_mask_ = np.array([[1, 0, 1], [0, 1, 0], [1, 0, 1]])
+        wet_full = np.tile(wet_mask_, (total_time_steps, levels, 1, 1))
+
+        # Even thetao, odd hfds for every time step
+        # Ex, timestep 0: thetao = 0, hfds = 1
+        # Ex, timestep 1: thetao = 2, hfds = 3
+        # Ex, timestep 2: thetao = 4, hfds = 5
+        # ...
```
Comment (jder) [inline tests/test_utils_data.py:212]: We don't actually care about this structure, right? I assume this is just copy-pasted? Maybe once we've gotten rid of multitons it will be easier to share this kind of setup between tests.

File: tests/test_utils_data.py [hunk 9]
```diff
@@ -186,3 +190,104 @@ def test_unnormalize_prognostic_tensor(normalize_input, fill_value):
...
+                "time": np.arange(total_time_steps),
+                "lev": DEPTH_LEVELS,
+                "lat": np.arange(lats),
+                "lon": np.arange(lons),
+            },
+        )
+        data_mean = data.mean() * 0.0
+        data_std = data.std() * 0.0 + 1.0
+        data, data_mean, data_std = validate_data(data, data_mean, data_std)
+        wet, wet_surface = extract_wet_mask(data, tensor_map.prognostic_var_names, hist)
```
Comment (jder) [inline tests/test_utils_data.py]: wet/wet_surface are unused.
```suggestion
```

**Review by jder — APPROVED** [review]
Just some questions/suggestions from me but generally makes sense!

File: tests/test_utils_data.py (hunk 8 above)
Comment (suryadheeshjith) [reply tests/test_utils_data.py:212]: Yes

File: src/ocean_emulators/aggregator/train.py (hunk 2 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/aggregator/train.py]: Fixed

File: src/ocean_emulators/utils/data.py (hunk 3 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/utils/data.py:211]: We are using data_reshaped for both the normalized (renamed to data_normalized) and unnormalized dicts. I treated it as a "just in case" clone

File: src/ocean_emulators/utils/data.py (hunk 4 above)
Comment (suryadheeshjith) [reply src/ocean_emulators/utils/data.py:212]: I will rename the function since we use them only for aggregators
